cmd/nfs-subdir-external-provisioner/provisioner.go

```diff
@@ -85,31 +83,27 @@ func (p *nfsProvisioner) Provision(ctx context.Context, options controller.Provi
 	}
 	glog.V(4).Infof("nfs provisioner: VolumeOptions %v", options)
 
-	pvcNamespace := options.PVC.Namespace
-	pvcName := options.PVC.Name
-
-	pvName := strings.Join([]string{pvcNamespace, pvcName, options.PVName}, "-")
-
 	metadata := &pvcMetadata{
 		data: map[string]string{
-			"name":      pvcName,
-			"namespace": pvcNamespace,
+			"name":      options.PVC.Name,
+			"namespace": options.PVC.Namespace,
+			"pvname":    options.PVName,
 		},
 		labels:      options.PVC.Labels,
 		annotations: options.PVC.Annotations,
 	}
 
-	fullPath := filepath.Join(mountPath, pvName)
-	path := filepath.Join(p.path, pvName)
-
 	pathPattern, exists := options.StorageClass.Parameters["pathPattern"]
-	if exists {
-		customPath := metadata.stringParser(pathPattern)
-		if customPath != "" {
-			path = filepath.Join(p.path, customPath)
-			fullPath = filepath.Join(mountPath, customPath)
-		}
+	if !exists {
+		pathPattern = defaultPathPattern
+	}
+
+	parsedPath := metadata.stringParser(pathPattern)
+	if parsedPath == "" {
```

Comment:
empty parsedPath means that the `stringParser()` necessarily failed? or pathPattern may exist but empty as well?
and what can cause stringParser to fail, pathPattern such as `foo-${.nonexistingvar}`? in such case `stringParser()` will return empty string?

Reply:
the `stringParser()` will return "" only if none of the patterns match. The tricky part is that if the pattern is `${.nonexisting1}-${.nonexisting1}` the parsedPath will be `-`, and will not be captured by the error and all nfs paths will be mapped to just `-`